cupy/indexing/indexing.py

```diff
@@ -76,7 +76,28 @@ def choose(a, choices, out=None, mode='raise'):
     return a.choose(choices, out, mode)
 
 
-# TODO(okuta): Implement compress
+def compress(condition, a, axis=None, out=None):
+    """Return selected slices of an array along given axis.
```

Comment:
Let's document that this function may synchronize, similarly to:

https://github.com/cupy/cupy/blob/1dc5e6deb8faa8157551d46ecbf5de5ede58a69d/cupy/core/core.pyx#L712-L714